hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -381,20 +384,30 @@ public static SparkConf registerClasses(SparkConf conf) {
     }
   }
 
+  private BulkInsertMapFunction<T> getBulkInsertMapFunction(
```

Comment:
I was looking at this for a different JIRA. but having a class for BulkInsertMapfunction seems like an overkill? Can we just replace with lambdas and kill the class ?

eg, in the code on master, just change to below

```
 JavaRDD<WriteStatus> writeStatusRDD = repartitionedRecords
        .mapPartitionsWithIndex((partition, recordItr) ->
            new CopyOnWriteLazyInsertIterable<>(recordItr, config, commitTime, table, fileIDPrefixes.get(partition)), true)
        .flatMap(List::iterator);

```

if you agree, just do it in the PR.. that way I wont introduce an unnecessary rebase for you :)

Reply:
cc @yihua

Reply:
For some subclasses in this PR, there are more complex logic in `BulkInsertMapFunction`, e.g., `BulkInsertMapFunctionForNonSortedRecords`, wondering if we should keep the API for now?

Reply:
Lets do whats easier and gets us moving faster .. :)